assets/src/stories-editor/components/video-block-edit-with-preview.js

```diff
@@ -0,0 +1,246 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { getBlobByURL, isBlobURL } from '@wordpress/blob';
+import {
+	Component,
+	createRef,
+} from '@wordpress/element';
+import {
+	BaseControl,
+	Button,
+	IconButton,
+	PanelBody,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+import {
+	BlockControls,
+	InspectorControls,
+	MediaUpload,
+	MediaUploadCheck,
+	RichText,
+} from '@wordpress/block-editor';
+
+const ALLOWED_MEDIA_TYPES = [ 'video' ];
+const VIDEO_POSTER_ALLOWED_MEDIA_TYPES = [ 'image' ];
+
+/**
+ * Mainly forked from the Core Video block Edit component, but allows the <video> to play instead of being disabled.
+ *
+ * There are very few changes from the Core Video block's components.
+ * The main change is that in render(), the <video> is not wrapped in <Disabled>, so it can play.
+ *
+ * @class
+ */
+class VideoBlockEditWithPreview extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editing: ! this.props.attributes.src,
+		};
+
+		this.videoPlayer = createRef();
+		this.posterImageButton = createRef();
+		this.toggleAttribute = this.toggleAttribute.bind( this );
+		this.onSelectURL = this.onSelectURL.bind( this );
+		this.onSelectPoster = this.onSelectPoster.bind( this );
+		this.onRemovePoster = this.onRemovePoster.bind( this );
+		this.onUploadError = this.onUploadError.bind( this );
+	}
+
+	componentDidMount() {
+		const {
+			attributes,
+			mediaUpload,
+			noticeOperations,
+			setAttributes,
+		} = this.props;
+		const { id, src = '' } = attributes;
+		if ( ! id && isBlobURL( src ) ) {
+			const file = getBlobByURL( src );
+			if ( file ) {
+				mediaUpload( {
+					filesList: [ file ],
+					onFileChange: ( [ { url } ] ) => {
+						setAttributes( { src: url } );
+					},
+					onError: ( message ) => {
+						this.setState( { editing: true } );
+						noticeOperations.createErrorNotice( message );
+					},
+					allowedTypes: ALLOWED_MEDIA_TYPES,
+				} );
+			}
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.attributes.poster !== prevProps.attributes.poster ) {
+			this.videoPlayer.current.load();
+		}
+	}
+	toggleAttribute( attribute ) {
+		return ( newValue ) => {
+			this.props.setAttributes( { [ attribute ]: newValue } );
+		};
+	}
+
+	onSelectURL( newSrc ) {
+		const { attributes, setAttributes } = this.props;
+		const { src } = attributes;
+
+		// Set the block's src from the edit component's state, and switch off
+		// the editing UI.
+		if ( newSrc !== src ) {
+			// Omit the embed block logic, as that didn't seem to work.
```

Comment:
Before, entering an embed URL below 'Insert from URL' was supposed to create an Embed block:

<img width="591" alt="insert-from-url" src="https://user-images.githubusercontent.com/4063887/61096357-d882c880-a41c-11e9-8556-0e60ff9cab87.png">

This is [intentionally disabled](https://github.com/ampproject/amp-wp/issues/2455) for YouTube, but even other embeds [like Vimeo](https://vimeo.com/111847786) don't seem to display.

**Is it OK that I removed the logic do convert to embeds?**

https://github.com/WordPress/gutenberg/blob/cb718ffdff57e35f05f4239ff03ceb476ae69a86/packages/block-library/src/video/edit.js#L111-L117

Entering a non-embed URL still works. But embed support will probably require copying a lot of other dependencies, like [createUpgradedEmbedBlock()](https://github.com/WordPress/gutenberg/blob/cb718ffdff57e35f05f4239ff03ceb476ae69a86/packages/block-library/src/embed/util.js#L61-L122), making this file much longer.

Reply:
> Is it OK that I removed the logic to convert to embeds?

I suppose so. AMP Stories doesn't really support any external providers AFAIK.

cc @westonruter

Reply:
AMP Story should support Twitter embed in theory.

Reply:
Pasting a tweet URL in a video block and converting it to a twitter embed would be a bit unexpected, no?

Reply:
True 😄 My comment was a bit out of context.

Reply:
External video providers aren't supported yet, but support is planned soon: https://github.com/ampproject/amphtml/issues/15847